deploy/quickstart.sh

```diff
@@ -1,72 +1,71 @@
-openssl req -new -x509 -batch -nodes -days 11000 \
-        -subj "/O=io.interconnectedcloud/CN=qdr-white.sa-telemetry.svc.cluster.local" \
-        -out qdr-server-certs/tls.crt \
-        -keyout qdr-server-certs/tls.key
-oc create secret tls qdr-white-cert \
-    --cert=qdr-server-certs/tls.crt \
-    --key=qdr-server-certs/tls.key \
-    --dry-run=true -o yaml | oc apply -f -
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: amq7-cert-manager
+  namespace: openshift-operators
+spec:
+  channel: alpha
+  installPlanApproval: Automatic
+  name: amq7-cert-manager
+  source: redhat-operators
+  sourceNamespace: openshift-marketplace
+  startingCSV: amq7-cert-manager.v1.0.0
 
-# generate certificates for ElasticSearch
-WORKING_DIR=./es-certs NAMESPACE="sa-telemetry" ./cert_generation.sh
-cd es-certs
+---
 
-# ref: https://github.com/openshift/cluster-logging-operator/blob/master/pkg/k8shandler/logstore.go#L76-L90
-oc create secret generic elasticsearch \
-    --from-file=elasticsearch.key \
-    --from-file=elasticsearch.crt \
-    --from-file=logging-es.key \
-    --from-file=logging-es.crt \
-    --from-file=admin-key=system.admin.key \
-    --from-file=admin-cert=system.admin.crt \
-    --from-file=admin-ca=ca.crt \
-    --dry-run=true -o yaml | oc apply -f -
+apiVersion: operators.coreos.com/v1alpha2
+kind: OperatorGroup
+metadata:
+  name: ${SAF_PROJECT}-og
+  namespace: ${SAF_PROJECT}
+spec:
+  targetNamespaces:
+  -  ${SAF_PROJECT}
 
-cd ..
+---
 
-# build out manifests
-ansible-playbook \
-    -e "registry_path=$(oc registry info)" \
-    -e "imagestream_namespace=$(oc project --short)" \
-    -e "prometheus_pvc_storage_request=2G" \
-    deploy_builder.yml
-
-# import container images
-if [[ "$*" =~ --downstream-secret=(.*) ]]; then
-    echo "Importing containers from downstream"
-    eval secret_file="${BASH_REMATCH[1]}"  # eval is for ~ expansion
-    # shellcheck disable=SC2154
-    oc create -f "${secret_file}"
-    ./import-downstream.sh
-else
-    echo "Importing containers from upstream"
-    ./import-upstream.sh
-fi
-
-# deploy the environment (requires interaction)
-./deploy.sh
-
-# teardown the environment when done (requires interaction)
-echo "Your environment should now be ready."
-echo "-- To teardown your environment, run:   ./deploy.sh DELETE"
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: service-assurance-operator
+  namespace: ${SAF_PROJECT}
+spec:
+  channel: alpha
+  installPlanApproval: Automatic
+  name: service-assurance-operator
+  source: redhat-service-assurance-operators
+  sourceNamespace: openshift-marketplace
+  startingCSV: service-assurance-operator.v0.1.0
+EOF
+while ! oc get csv | grep service-assurance-operator | grep Succeeded; do echo waiting; done
+oc create -f - <<EOF
+apiVersion: infra.watch/v1alpha1
+kind: ServiceAssurance
+metadata:
+  name: saf-default
+  namespace: ${SAF_PROJECT}
+spec:
```

Comment:
I recently changed this `spec` to account for how Ansible does not merge dictionaries very well, so we'll end up with all sorts of issues in the future if we build out more complex dictionaries with defaults, and we don't want to have to define the entire dictionary to override a single value.

So I've flattened the pattern. You'll want to migrate this to:

```
metricsEnabled: true
eventsEnabled: true
```

Reply:
Ah yes, I saw that, thanks.